stdlib/SuiteSparse/src/cholmod.jl

```diff
@@ -397,56 +397,51 @@ Factor(FC::FactorComponent) = Factor(FC.F)
 #################
 
 # Dense wrappers
-## Note! Integer type defaults to Cint, but this is actually not necessary, but
-## making this a choice would require another type parameter in the Dense type
 
 ### cholmod_core_h ###
-function allocate_dense(nrow::Integer, ncol::Integer, d::Integer, ::Type{Float64})
-    Dense(ccall((:cholmod_l_allocate_dense, :libcholmod), Ptr{C_Dense{Float64}},
-        (Csize_t, Csize_t, Csize_t, Cint, Ptr{Cvoid}),
-        nrow, ncol, d, REAL, common_struct))
-end
-function allocate_dense(nrow::Integer, ncol::Integer, d::Integer, ::Type{Complex{Float64}})
-    Dense(ccall((:cholmod_l_allocate_dense, :libcholmod), Ptr{C_Dense{Complex{Float64}}},
-        (Csize_t, Csize_t, Csize_t, Cint, Ptr{Cvoid}),
-        nrow, ncol, d, COMPLEX, common_struct))
+function allocate_dense(m::Integer, n::Integer, d::Integer, ::Type{Tv}) where {Tv<:VTypes}
+    Dense(ccall((@cholmod_name("allocate_dense"), :libcholmod), Ptr{C_Dense{Tv}},
+                (Csize_t, Csize_t, Csize_t, Cint, Ptr{Cvoid}),
+                m, n, d, xtyp(Tv), common_struct))
 end
 
-free_dense!(p::Ptr{C_Dense{T}}) where {T} = ccall((:cholmod_l_free_dense, :libcholmod),
-    Cint, (Ref{Ptr{C_Dense{T}}}, Ptr{Cvoid}), p, common_struct)
-
-function zeros(m::Integer, n::Integer, ::Type{T}) where T<:VTypes
-    Dense(ccall((:cholmod_l_zeros, :libcholmod), Ptr{C_Dense{T}},
-        (Csize_t, Csize_t, Cint, Ptr{UInt8}),
-         m, n, xtyp(T), common_struct))
+function free!(p::Ptr{C_Dense{Tv}}) where {Tv<:VTypes}
```

Comment:
The idea was to use the CHOLMOD name of the function for the wrappers to make it easy to use map between the C API and the wrappers. Then alongside to have a "julia API" based on the wrapper functions. Not necessarily saying that is the better way to do it, but just wanted to mention the motivation.

Reply:
ah, fair enough. This just simplified a bit of later code, but we can change it back if you would prefer.

Reply:
Keep it as it is in the PR? or as it is in the current code? (I don't mind either way)

Reply:
As it is in the PR.